fusionSuite/frontend
# tec: Add e2e tests (cypress)

Changes proposed in this pull request:

- Add and configure Cypress
- Add two commands to the Makefile (`make e2e-open` and `make e2e-run`)
- Configure the CI to execute the Cypress tests
    - it basically installs backend and frontend side-by-side, with Nginx as web server
    - then, it run the Cypress GitHub action on the frontend
    - the old `ci.yml` has been renamed `tests.yml`
- Add tests for login, organizations (list, creation and deletion) and users (list, creation and deletion)
- Remove the old Angular tests which are no longer useful (e2e tests should catch the same issues… and more)
- Unit test the OrganizationsSorter (the CI was failing because there were no tests any more to execute and the OrganizationsSorter is complex enough to be tested)

How to test the feature manually:

1. Update your dependencies with `make install`
2. Open Cypress with `make e2e-open` or run the tests with `make e2e-run`
3. Check that everything works fine

More information on how to configure e2e tests at: https://github.com/fusionSuite/documentation/pull/28

Pull request checklist:


_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._

Related to https://github.com/fusionSuite/documentation/pull/28
